source/contracts/libraries/Delegator.sol

```diff
@@ -24,13 +24,7 @@ contract Delegator is DelegationTarget {
             //0x40 is the address where the next free memory slot is stored in Solidity
             let _calldataMemoryOffset := mload(0x40)
             // Update the pointer at 0x40 to point at new free memory location so any theoretical allocation doesn't stomp our memory in this call
-            let _size := 0
-            switch gt(calldatasize, 32)
-            case 1 {
-                _size := calldatasize
-            } default {
-                _size := 32
-            }
+            let _size := and(add(calldatasize, 0x1f), not(0x1f))
```

Comment:
Wat?  `not(0x1f)` is a constant isn't it?  Or does `not` take in a memory address?  What is at memory address `0x1f` in that case?  Either way, it feels like we should have a comment indicating what the magic `0x1f` is.  Same question for `add`, does it take in two memory addresses and adds them together?  If so, does `and` then behave different from those two and instead takes a value directly (returned from `add`/`not`) or do `add`/`not` return memory addresses.  If so, do they use the 0x40 pointer to get a memory address in which case we need to consider that our pointer is moving while we do this?

Reply:
`0x1f` is a constant. This is the padding value suggested by the audit to use.

`not`,`and`, and `add` in this line are simply modifying the size of memory we are reserving. They are not intended to be memory addresses.

Reply:
I'll note in a comment that we are including padding in the size calculation